### ISSUE TITLE
Unable to freeze TimeDistributed(Dense()) Layer 

### ISSUE THREAD

**Issue by justiceamoh** [keras-team/keras#7504]
When you pass a `trainable=False` to a `TimeDistributed(Dense())` layer, the layer still remains trainable. I think this is because the option is not passed down from the `TimeDistributed` wrapper to the embedded `Dense` layer. It's also impossible to directly access the Dense layer within the wrapper to set the `trainable` option on there. Any suggestions? 

To recreate, just use this:

```python
model = Sequential()
model.add(InputLayer(input_shape=(tsteps,nfeats),name='Input'))
model.add(TimeDistributed(Dense(100,activation='relu')))
model.add(TimeDistributed(Dense(100,activation='relu')))
model.add(Flatten())
model.add(Dense(nclass, activation='softmax'))

for layer in model.layers:
      layer.trainable=False

model.compile(loss='categorical_crossentropy',optimizer='adam')
model.summary()
```

**Comment by Dref360** [comment]
Please see : https://github.com/fchollet/keras/pull/7145

**Comment by mahnerak** [comment]
1. The inner Dense layer of TimeDistributed layers is accessible via attribute named "layer".
```python
for layer in model.layers:
    if hasattr(layer, 'layer'):
        layer.layer.trainable = False
    layer.trainable = False
```

2. When you are using non-masked inputs, simple `Dense` layer can replace `TimeDistributed(Dense())`.

3. However, I assume this as a bug. Also the wrapper Bidirectional() may be affected.

**Comment by justiceamoh** [comment]
Thanks! Actually, answer (1) addressed my issue.

**Comment by mahnerak** [comment]
@justiceamoh I recommend that you leave this issue opened, because as we can see, wrapper layers like TimeDistributed and Bidirectional doesn't support `trainable` argument proper propagation.
I'll try to fix that and create a pull-request later.

**Comment by justiceamoh** [comment]
Reopening then. Thanks!

**Comment by ylmeng** [comment]
@mahnerak Is there any progress?
I trained a model and want to use a part of it in another model. However, since TimeDistributed is involved, I cannot make that part trainable, though it works if I load it as trainable=False. I am not completely sure if TimeDistributed caused the problem, but it seems to be a common problem.

**Comment by gerwin3** [comment]
This was actually fixed with https://github.com/keras-team/keras/commit/c7efb4e30f914555954f1d07c585c7a2ff9150de
